osbenchmark/resources/custom-test-procedures.json.j2

```diff
@@ -0,0 +1,58 @@
+{
+  "name": "custom-test-procedures",
+  "description": "Customized test procedure with custom operations generated by create-workload feature in OpenSearch Benchmark.",
+  "default": true,
+  "schedule": [
+    {
+      "operation": "delete-index"
+    },{% raw %}
+    {
+      "operation": {
+        "operation-type": "create-index",
+        "settings": {{index_settings | default({}) | tojson}}
+      }
+    },{% endraw %}
+    {
+      "operation": {
+        "operation-type": "cluster-health",
+        "index": {{ indices | map(attribute='name') | list | join(',') | tojson }},{% raw -%}
+        "request-params": {
+          "wait_for_status": "{{cluster_health | default('green')}}",
+          {% endraw -%}
+          "wait_for_no_relocating_shards": "true"
+        },
+        "retry-until-success": true
+      }
+    },
+    {
+      "operation": "index-append",
+      {% raw -%}"clients": {{bulk_indexing_clients | default(8)}},
+      "ignore-response-error-level": "{{error_level | default('non-fatal')}}"
+    },
+    {% endraw -%}
+    {
+      "name": "refresh-after-index",
+      "operation": "refresh"
+    },
+{%- block queries -%}
+    {% for query in custom_queries %}
+    {
+      "operation":"{{query.name}}",
+      {% raw -%}
+      "warmup-iterations": {{warmup_iterations | default(50)}},
+      "iterations": {{iterations | default(100)}}
+      {%- if not target_throughput %}
+      ,"target-throughput": 3
+      {%- elif target_throughput is string and target_throughput.lower() == 'none' %}
+      {%- else %}
+      ,"target-throughput": {{ target_throughput | tojson }}
+      {%- endif %}
+      {%-if search_clients is defined and search_clients %}
+      ,"clients": {{ search_clients | tojson}}
```

Comment:
We should change this (later) along with all the existing workloads to use the following pragma instead as in the `pmc`workload:
```
          "target-throughput": {{ target_throughput | default(20) | tojson }},
          "clients": {{ search_clients | default(1) }}
```

Reply:
You're right, it'd be much cleaner with that approach. I'll add this structure and test it out.